idx/memory/memory.go

```diff
@@ -269,6 +374,101 @@ func (m *MemoryIdx) GetPath(orgId int, path string) []idx.Archive {
 	return archives
 }
 
+func (m *MemoryIdx) Tag(orgId int, tag string, from int64) map[string]uint32 {
+	if !tagSupport {
+		log.Warn("memory-idx: received tag query, but tag support is disabled")
+		return nil
+	}
+
+	m.RLock()
+	defer m.RUnlock()
+
+	var tags TagIndex
+	var ok bool
+	if tags, ok = m.Tags[orgId]; !ok {
+		return nil
+	}
+
+	result := make(map[string]uint32)
+
+	for value, ids := range tags[tag] {
+		valueCnt := uint32(0)
+		for id := range ids {
+			var def *idx.Archive
+			var ok bool
+			if def, ok = m.DefById[id.String()]; !ok {
+				// should never happen because every ID that is in the tag index
+				// must be present in the byId lookup table
+				CorruptIndex.Inc()
+				log.Error(3, "memory-idx: ID %s is in tag index but not in the byId lookup table", id.String())
+				continue
+			}
+
+			if def.LastUpdate < from {
+				continue
+			}
+
+			valueCnt++
+		}
+		if valueCnt > 0 {
+			result[value] = valueCnt
+		}
+	}
+
+	return result
+}
+
+func (m *MemoryIdx) TagList(orgId int) []string {
+	if !tagSupport {
+		log.Warn("memory-idx: received tag query, but tag support is disabled")
+		return nil
+	}
+
+	m.RLock()
+	defer m.RUnlock()
+
+	var tags TagIndex
+	var ok bool
+	if tags, ok = m.Tags[orgId]; !ok {
+		return nil
+	}
+
+	results := make([]string, 0, len(tags))
+
+	for k := range tags {
+		results = append(results, k)
+	}
+
```

Comment:
I think you can replace these 4 lines with:

```
results := make([]string, len(tags))
copy(results, tags)
```

Reply:
nice